GNUmakefile

```diff
@@ -5,7 +5,20 @@ NAMESPACE=edu
 NAME=biganimal
 BINARY=terraform-provider-${NAME}
 VERSION=0.3.1
-OS_ARCH=darwin_amd64
+
+# Figure out the OS and ARCH of the
+# builder machine
+os       ?= $(shell uname|tr A-Z a-z)
+ifeq ($(shell uname -m),x86_64)
+  arch   ?= amd64
+endif
+ifeq ($(shell uname -m),i686)
+  arch   ?= 386
+endif
+ifeq ($(shell uname -m),aarch64)
```

Comment:
Hi Serdar, thank you for fixing this. I did a test on my M1 MacBook.  

```
kunliu@LAPTOP678-MA-US 0.3.1 % uname -m
arm64
```

Perhaps we can add one more IF condition here ?
```
ifeq ($(shell uname -m),arm64)
  arch   ?= arm64
endif
```

Reply:
That's a very good point, @kunliuedb, thanks for pointing out! 
The funny thing is, both `arm64` and `aarch64` technically should be the same thing, more info [here](https://en.wikipedia.org/wiki/AArch64). 
And our release workflow [generates](https://github.com/EnterpriseDB/terraform-provider-biganimal/releases/tag/v0.0.0-release) `arm64` executables, instead of `aarch64`. 

So, maybe we may not need `aarch64` at all, but I'm not sure. That's why, I'm only going to add `arm64` and not delete `aarch64`.